yax/artifacts/coverage_data.py

```diff
@@ -0,0 +1,67 @@
+from yax.state.type import Artifact
+from os import listdir
+from os.path import isfile, join, splitext
```

Comment:
stylistically, these functions are not placed directly in the namespace. it's longer, but usually you see `os.path.join`

seeing `join` on it's own is unusual.